### PR TITLE
Update stub for "ibm_db2"

### PR DESCRIPTION
**Pull request by phansys**

### CODE REVIEW

**Comment by phansys** [comment]
Is there something I can do to [fix the CI](https://github.com/JetBrains/phpstorm-stubs/actions/runs/4258194218/jobs/7535108114#step:4:148)?
```
Fatal error: Uncaught PhpParser\Error: Syntax error, unexpected ']', expecting ')' on line 1212 in /opt/project/phpstorm-stubs/vendor/nikic/php-parser/lib/PhpParser/ParserAbstract.php:318
```

**Comment by isfedorov** [comment]
@phansys yes you can :) you can fix the typo on line 1212 as the failure is caused by it.

function db2_prepare($connection, string $statement, array $options = []**]**) {}

**Comment by phansys** [comment]
>function db2_prepare($connection, string $statement, array $options = []]) {}

:man_facepalming: 

Thank you @isfedorov!

File: ibm_db2/ibm_db2.php [hunk 1]
```diff
@@ -49,7 +49,7 @@
  * successful. If the connection attempt fails, db2_connect
  * returns false.
  */
-function db2_connect($database, $username, $password, array $options = null) {}
+function db2_connect(string $database, ?string $username, #[\SensitiveParameter] ?string $password, array $options = []) {}
```
Comment (isfedorov) [inline ibm_db2/ibm_db2.php:52]: Please update corresponding params in PHPDoc as well

File: ibm_db2/ibm_db2.php [hunk 2]
```diff
@@ -94,7 +94,7 @@ function db2_commit($connection) {}
  * password parameters. If the connection attempt fails,
  * db2_pconnect returns false.
  */
-function db2_pconnect($database, $username, $password, array $options = null) {}
+function db2_pconnect(string $database, ?string $username, #[\SensitiveParameter] ?string $password, array $options = []) {}
```
Comment (isfedorov) [inline ibm_db2/ibm_db2.php:97]: Please update PHPDoc as well

File: ibm_db2/ibm_db2.php [hunk 3]
```diff
@@ -661,7 +661,7 @@ function db2_primarykeys() {}
  * </td>
  * </tr>
  */
-function db2_procedure_columns($connection, $qualifier, $schema, $procedure, $parameter) {}
+function db2_procedure_columns($connection, ?string $qualifier, string $schema, string $procedure, ?string $parameter) {}
```
Comment (isfedorov) [inline ibm_db2/ibm_db2.php:664]: please update PHPDoc for `$parameter`

File: ibm_db2/ibm_db2.php [hunk 4]
```diff
@@ -1024,24 +1024,24 @@ function db2_specialcolumns() {}
  * </tr>
  * </table>
  */
-function db2_statistics($connection, $qualifier, $schema, $table_name, $unique) {}
+function db2_statistics($connection, ?string $qualifier, ?string $schema, string $table_name, bool|int $unique) {}
```
Comment (isfedorov) [inline ibm_db2/ibm_db2.php:1027]: Please update PHPDoc here as well

File: ibm_db2/ibm_db2.php [hunk 5]
```diff
@@ -1628,27 +1628,27 @@ function db2_setoption() {}
  * <p>
  * Returns false if no row was retrieved.
  */
-function db2_fetch_object($stmt, $row_number = null) {}
+function db2_fetch_object($stmt, int $row_number = null): stdClass|false {}
```
Comment (isfedorov) [inline ibm_db2/ibm_db2.php:1631]: Does the funtion really returns stdClass or just a "some" object? Maybe `object` type hint would be better here?

**Review by isfedorov — CHANGES_REQUESTED** [review]
@phansys Thank you for the contribution! Overall PR is good to go, I've left just some comments regarding not updated PHPDocs and a questions regarding stdClass as a return type. Please find them inline

File: ibm_db2/ibm_db2.php (hunk 5 above)
Comment (phansys) [reply ibm_db2/ibm_db2.php:1631]: All the `db_*()` functions returning objects are returning instances of `stdClass`. See php/pecl-database-ibm_db2#34.

File: ibm_db2/ibm_db2.php (hunk 4 above)
Comment (phansys) [reply ibm_db2/ibm_db2.php:1031]: Indeed. I must check this one in the extension. Thank you!
https://github.com/php/pecl-database-ibm_db2/blob/f4bfd692677db26cb16b5c3e72e0e61f122f05ac/ibm_db2.c#L3718

Comment (phansys) [reply ibm_db2/ibm_db2.php]: Updated. See php/pecl-database-ibm_db2#50.

File: ibm_db2/ibm_db2.php (hunk 1 above)
Comment (phansys) [reply ibm_db2/ibm_db2.php:43]: Done.

**Pull request merged**